### PR TITLE
Add support for all UARTS on STM32

### PR DESCRIPTION
**Pull request by jeremywood**
Adds support for configuring UART3-UART7, increasing the maximum number of configurable UARTs on STM32 from 3 to 8.

### CODE REVIEW

File: hw/mcu/stm/stm32_common/src/hal_uart.c [hunk 1]
```diff
@@ -63,6 +63,37 @@ uart_by_port(int port)
...
+    if (port == 6) {
+        return &uarts[index];
+    }
+    index++;
+#endif
+#if MYNEWT_VAL(UART_7)
+    if (port == 7) {
+        return &uarts[index];
+    }
+    index++;
```
Comment (utzig) [inline hw/mcu/stm/stm32_common/src/hal_uart.c:96]: No need to increment here.

**Comment by kasjer** [comment]
Quick check of number of UARTs/USARTs on ST page reveals that STMF423 and STMF413 (that has BSP in Mynewt) have 10 of them. While I don't have problem with this code PR, title is misleading suggesting that all UARTs are covered.
Since there are two more UARTS out there, way not cover it all here and we could skip next PR with additional 2 in the future.

File: hw/mcu/stm/stm32_common/src/hal_uart.c (hunk 1 above)
Comment (jeremywood) [reply hw/mcu/stm/stm32_common/src/hal_uart.c:96]: Fixed.

**Comment by jeremywood** [comment]
@kasjer Good catch on the STMF4XX support for 10 UARTS. I've searched across all of the STM headers, and that does appear to be the highest port count.

**Pull request merged**